src/components/ComponentView.vue

```diff
@@ -1,38 +1,73 @@
-<template>
-    <div>
-        <component v-if="componentData" :is="componentData.component" v-bind="componentData.data" :style="componentData.rootCss" class="instance" v-on="componentData.listeners"></component>
-    </div>
-</template>
-
 <script>
 import '@/stateless/define_helpers'
-import components from '../components.js'
+import SelectProps from './SelectProps.vue'
+import * as library from '../library.js'
+import { getFlatUsecases, getProps } from '../component_utils'
 
 export default {
+    components: {
+        SelectProps,
+    },
     data () {
         return {
             name: null,
             value: null,
+            query: {},
         }
     },
     computed: {
         componentData () {
             if (this.name === null)
                 return null
 
-            let data = components[this.name]
+            let component = library[this.name]
+
+            let modelName = component.model ? component.value : 'value'
+            let modelEvent = component.model ? component.event : 'input'
```

Comment:
All of those can be `const`.
I see this wasn't introduced here but we can still fix it as we go